src/lib/types.ts

```diff
@@ -124,10 +124,14 @@ export abstract class PostgrestBuilder<T> implements PromiseLike<PostgrestRespon
           count = parseInt(contentRange[1])
         }
       } else {
-        error = await res.json()
-
-        if (error && this.shouldThrowOnError) {
-          throw error
+        try {
+          error = await res.json()
+        } catch (e) {
+          error = await res.text()
+        } finally {
+          if (error && this.shouldThrowOnError) {
+            throw error
+          }
```

Comment:
We can't reuse the response body since it was consumed with `res.json()`. Also, we want the error to be an object with a `message` property. So how about:

```js
        const body = await res.text()

        try {
          error = JSON.parse(body)
        } catch {
          error = {
            message: body
          }
        }

        if (error && this.shouldThrowOnError) {
          throw error
        }
```